src/Console/Commands/JsonApiGeneratorCommand.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace CloudCreativity\LaravelJsonApi\Console\Commands;
+
+use Illuminate\Console\GeneratorCommand as LaravelGeneratorCommand;
+use Illuminate\Filesystem\Filesystem;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputArgument;
+
+abstract class JsonApiGeneratorCommand extends LaravelGeneratorCommand
+{
+    /**
+     * The console command name.
+     *
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description;
+
+    /**
+     * The type of class being generated.
+     *
+     * @var string
+     */
+    protected $type;
+
+    /**
+     * Whether the resource type is non-dependant on eloquent
+     *
+     * @var mixed
+     */
+    protected $isIndependant = false;
+
+    /**
+     * Create a new config clear command instance.
+     *
+     * @return void
+     */
+    public function __construct(Filesystem $files)
+    {
+        parent::__construct($files);
+
+        $this->subNamespace = config('json-api.generator.namespace', 'JsonApi');
+        $this->useEloquent = config('json-api.generator.use_eloquent', true);
+    }
+
+    /**
+     * Build the class with the given name.
+     *
+     * Remove the base controller import if we are already in base namespace.
+     *
+     * @param  string  $name
+     * @return string
+     */
+    protected function buildClass($name)
+    {
+        $namespace = $this->getNamespace($name);
+
+        $stub = $this->files->get($this->getStub());
+
+        $this->replaceNamespace($stub, $name)
+                ->replaceResourceType($stub, $this->getPluralizedResource());
+
+        return $stub;
+    }
+
+    /**
+     * Replace the value of the resource type constant
+     *
+     * @param mixed $stub
+     * @param mixed $resource
+     */
+    protected function replaceResourceType(&$stub, $resource)
+    {
+        $stub = str_replace(
+            'dummyResourceType', camel_case($resource), $stub
+        );
+
+        return $this;
+    }
+
+    /**
+     * Get the plural version of the resource name
+     *
+     * @return string
+     */
+    protected function getPluralizedResource()
+    {
+        return str_plural($this->getResourceName());
+    }
+
+    /**
+     * Get the resource name
+     *
+     * @return string
+     */
+    protected function getResourceName()
+    {
+        return ucwords($this->argument('resource'));
+    }
+
+    /**
+     * Get the desired class name from the input.
+     *
+     * @return string
+     */
+    protected function getNameInput()
+    {
+        return trim($this->type);
+    }
+
+    /**
+     * Get the stub file for the generator.
+     *
+     * @return string
+     */
+    protected function getStub()
+    {
+        if($this->isIndependant) {
+            return __DIR__.'/stubs/'.strtolower($this->type).'.stub';
```

Comment:
Can we move the `stubs` directory to the root project folder. I have a strong belief that the `src` folder should just be for classes!